run_headless_sim.sh

```diff
@@ -0,0 +1,325 @@
+NUM_ROVERS=$2
+echo "The user requested $NUM_ROVERS rovers."
+if [[ $NUM_ROVERS -gt $MAX_ROVERS ]]; then
+    echo "User requested too many rovers. Maximum rovers is $MAX_ROVERS. Exiting."
+    exit 2
+fi
+
+#---------------------------------------------------------#
+
+addCollectionZone
+
+addGroundPlane
+
+
+#---------------------------------------------------------#
+
+# Add the rovers to the simulation
+
+#  The distances that determine the X, Y coords of the rovers is determined as follows:
+#  The distance to the rover from a corner position is calculated differently
+#  than the distance to a cardinal position.
+# 
+#  The cardinal direction rovers are a straightforward calculation where:
+#      a = the distance to the edge of the collection zone
+#          i.e., 1/2 of the collection zone square side length
+#      b = the 50cm distance required by the rules for placing the rover
+#      c = offset for the simulation for the center of the rover (30cm)
+#          i.e., the rover position is at the center of its body
+# 
+#  The corner rovers use trigonometry to calculate the distance where each
+#  value of d, e, and f, are the legs to an isosceles right triangle. In
+#  other words, we are calculating and summing X and Y offsets to position
+#  the rover.
+#      d = a
+#      e = xy offset to move the rover 50cm from the corner of the collection zone
+#      f = xy offset to move the rover 30cm to account for its position being
+#          calculated at the center of its body
+# 
+#                        *  *          d = 0.508m
+#                      *      *        e = 0.354m
+#                    *          *    + f = 0.212m
+#                  *     /*     *    ------------
+#                  *    / | f *            1.072m
+#                    * /--| *
+#                     /* *
+#                    / | e
+#                   /--|
+#      *************
+#      *          /|
+#      *         / |
+#      *        /  | d                 a = 0.508m
+#      *       /   |     *********     b = 0.500m
+#      *      /    |     *       *   + c = 0.300m
+#      *     *-----|-----*---*   *   ------------
+#      *        a  *  b  * c     *         1.308m
+#      *           *     *********
+#      *           *
+#      *           *
+#      *           *
+#      *************
+
+# Specify rover names
+ROVER_NAMES=( "achilles" "aeneas" "ajax" "diomedes" "hector" "paris" "thor" "zeus" )
+
+# Specify rover start coordinates
+ROVER_POSITIONS_X=( -1.308 0.000 1.308 0.000 1.072 -1.072 -1.072 1.072 )
+ROVER_POSITIONS_Y=( 0.000 -1.308 0.000 1.308 1.072 -1.072 1.072 -1.072 )
+ 
+# In this case, the yaw is the value that turns rover "left" and "right" */
+ROVER_YAWS=( 0.000 1.571 -3.142 -1.571 -2.356 0.785 -0.785 2.356 )
+
+echo "Adding rovers to Gazebo and starting their ROS nodes..."
+
+# Add rovers to the simulation and start the associated ROS nodes
+for (( i=0;i<=$NUM_ROVERS;i++ ));
```

Comment:
Is this correct, or should it be `i<$NUM_ROVERS`?

Reply:
Yes that's a bug! There is a similar off by one error where I send the autonomous command.